lib/EventsGenerator.js

```diff
@@ -0,0 +1,174 @@
+      { type: String, description: 'relative path to the config yaml file' })
+    this.option('full-key-to-events-manifest', {
+      type: String,
+      description: 'key in config path that resolves to events manifest e.g. "application.events"',
+      default: ''
+    })
+
+    this.configPath = this.destinationPath(this.options['config-path'])
+    this.fullKeyToRuntimeManifest = this.options['full-key-to-manifest']
+    this.actionFolder = this.options['action-folder'] // todo ensure this is relative to root
+    this.fullKeyToEventsManifest = this.options['full-key-to-events-manifest']
+    this.projectConfig = config.get('project')
+    this.providerMetadataToProviderMappingDotEnv = process.env.AIO_events_providermetadata_to_provider_mapping
+  }
+
+  async initEventsClient () {
+    if (!this.projectConfig) {
+      this.env.error('Incomplete .aio configuration, please import a valid Adobe Developer Console configuration via `aio app use` first.')
+    }
+    const orgCode = this.projectConfig.org.ims_org_id
+    const accessToken = await getToken(CLI)
+    const X_API_KEY = process.env.SERVICE_API_KEY
+    const eventsClient = await eventsSdk.init(orgCode, X_API_KEY, accessToken)
+    return eventsClient
+  }
+
+  async promptForEventsDetails (defaultValues, options) {
+    if (!this.options['skip-prompt']) {
+      const eventsClient = await this.initEventsClient()
+      const runtimeActionName = await promptForRuntimeAction(this)
+      let regName = defaultValues.regName
+      let regDesc = defaultValues.regDesc
+
+      const basicDetails = await this.prompt([
+        {
+          type: 'input',
+          name: 'regName',
+          message: 'We are about to create a new Event registration.\nHow would you like to name this registration?',
+          default: regName,
+          when: !this.options['skip-prompt']
+        },
+        {
+          type: 'input',
+          name: 'regDesc',
+          message: 'What is this registration being created for?',
+          default: regDesc,
+          when: !this.options['skip-prompt']
+        }
+      ])
+      regName = basicDetails.regName
+      regDesc = basicDetails.regDesc
+      const selectedProvidersToEventMetadata = await promptForEventsOfInterest(eventsClient, this, options)
+      return { regName, regDesc, selectedProvidersToEventMetadata, runtimeActionName }
+    }
+  }
+
+  /**
+   * Adds a new event registration to the project
+   *
+   * @param {string} eventDetails
+   * @param {object} [options={}]
+   * @param {object} [options.dotenvStub]
+   * @param {string} options.dotenvStub.label
+   * @param {Array<string>} options.dotenvStub.vars
+   * @param {object} [options.dependencies]
+   * @param {object} [options.devDependencies]
+   * @memberof EventsGenerator
+   */
+  addEvents (eventDetails, tplActionPath, options = {}) {
+    // NOTE: it's important to load a fresh manifest now, as we do want to include the
+    // latest written data
+    if (eventDetails) {
```

Comment:
it's a matter of preference, but I would return here if eventDetails is undefined/null than put all the logic in in a singular if block

Reply:
done